### PR TITLE
Infer IS (NOT) NULL as `boolean` instead of `boolean | null`

### PR DESCRIPTION
**Pull request by karlhorky**
Avoid the `| null` here, from [my comment](https://github.com/ts-safeql/safeql/issues/199#issuecomment-2356404472):

```ts
// 💥 Query has incorrect type annotation.
//	Expected: { ... is_mammal: boolean; }
//	Actual: { ... is_mammal: boolean | null; }[]
await sql<{ id: number, first_name: string, is_mammal: boolean }[]>`
  SELECT
    animals.id,
    animals.first_name,
    mammals.id IS NOT NULL AS is_mammal
  FROM
    animals
    LEFT JOIN mammals ON animals.id = mammals.animal_id
`;
```

The tests I added are passing, but I'm not sure if I am missing out on other cases which should NOT be non-nullable.

By the way, is there something like [AST explorer](https://astexplorer.net/) for the SQL AST types? I am fumbling a bit in the dark here, searching through the codebase - would be great to have a visual UI to select the parts of the code I'm interested in and have the node types identified.

### CODE REVIEW

File: packages/generate/src/utils/get-nonnullable-columns.ts [hunk 1]
```diff
@@ -195,6 +199,10 @@ function getNodeName(node: LibPgQueryAST.Node | undefined): string {
     return concatStringNodes(node.ColumnRef.fields);
   }
 
+  if (node?.NullTest) {
+    return getNodeName(node.NullTest.arg);
+  }
+
```
Comment (karlhorky) [inline packages/generate/src/utils/get-nonnullable-columns.ts:205]: not 100% sure if these lines are needed

**Comment by karlhorky** [comment]
One thing this doesn't touch is `IS NOT NULL` and `IS NULL` inside of a subquery, which yields `unknown`:

```sql
// 💥 Query has incorrect type annotation.
//	Expected: { is_null: boolean }[]
	  Actual: { is_null: unknown }[]
await sql<{ is_null: boolean }>`
  SELECT (SELECT 1 IS NULL) AS is_null
`
```

File: packages/generate/src/utils/get-nonnullable-columns.ts (hunk 1 above)
Comment (Newbie012) [reply packages/generate/src/utils/get-nonnullable-columns.ts:205]: I think it's okay for now to keep it as it is

**Review by Newbie012 — APPROVED** [review]

**Comment by Newbie012** [comment]
Welll done! LGTM. Regaring a AST Explorer - sadly there isn't. I usually rely heavily on logs and breakpoints.

**Review by Newbie012 — APPROVED** [review]

**Comment by karlhorky** [comment]
Nice, thanks for the review and merge!

> Regaring a AST Explorer - sadly there isn't. I usually rely heavily on logs and breakpoints.

Interesting, yeah then I'll work like this to start.

Wonder how easy it is to contribute something like this to AST explorer... 🤔 Would be an interesting thing to have the SQL AST more observable / visually debuggable. Or alternatively, build some minimal custom tooling. I'll keep it in mind.

**Comment by Newbie012** [comment]
@karlhorky https://github.com/Newbie012/pg-query-ast-explorer

**Comment by karlhorky** [comment]
Omg amazing, that looks great!! I will definitely use this for the next PRs!

https://pg-query-ast-explorer.vercel.app/

Screenshot 2024-09-20 at 15 04 31

**Pull request merged**